api_tests/registrations/views/test_registration_list.py

```diff
@@ -1377,6 +1391,160 @@ def test_assert_file_validity_on_registration_metadata(
             expect_errors=True)
         assert res.status_code == 201
 
+class TestRegistrationCreate(TestNodeRegistrationCreate):
+    """
+    Tests for creating registration through new workflow -
+    POST RegistrationList
+    """
+
+    @pytest.fixture()
+    def url_registrations(self, project_public):
+        return '/{}registrations/'.format(
+            API_BASE
+        )
+
+    @pytest.fixture()
+    def url_registrations_ver(self, project_public):
+        return '/{}registrations/?version={}'.format(
+            API_BASE,
+            CREATE_REGISTRATION_FIELD_CHANGE_VERSION
+        )
+
+    @pytest.fixture()
+    def payload(self, draft_registration):
+        return {
+            'data': {
+                'type': 'registrations',
+                'attributes': {
+                    'draft_registration': draft_registration._id,
+                    'registration_choice': 'immediate'
+                }
+            }
+        }
+
+    @pytest.fixture()
+    def payload_ver(self, draft_registration):
+        return {
+            'data': {
+                'type': 'registrations',
+                'attributes': {
+                    'draft_registration_id': draft_registration._id,
+                    'registration_choice': 'immediate'
```

Comment:
There are tests in TestNodeRegistrationCreate (which TestRegistrationCreate inherits) that do test using embargo_end_date on the new version. However, the API does not currently throw an error when you use the old fields on the new version. Is this something worth trying to add?

Reply:
(talked irl) i do think it's a problem, because if an api user tries specifying an embargo with `registration_choice` and `lift_embargo` in the latest api version, the registration will instead be made public immediately without warning.

but instead of fixing it in this pr, we'll do it in another ticket

Reply:
Ticket has been created to address this: https://openscience.atlassian.net/browse/ENG-1490